src/main/java/RecommendationsServlet.java

```diff
@@ -0,0 +1,184 @@
+  private VisualizationData getRandomRecommendation(String userId, ArrayList<VisualizationData> userHistory) {
+    ArrayList<VisualizationData> invalidRecommendations = new ArrayList<VisualizationData>();
+    boolean isValid = false;
+    while (!isValid) {
+      String randomPersonType = personType.get(getRandomInt(personType.size()));
+      String randomAction = action.get(getRandomInt(action.size()));
+      String randomLocation = location.get(getRandomInt(location.size()));
+      String randomYear = years.get(getRandomInt(years.size()));
+      VisualizationData randomRecommendation = new VisualizationData(
+          userId, randomPersonType, randomAction, randomLocation, randomYear);
+      if (isRecommendationValid(randomRecommendation, userHistory) && !invalidRecommendations.contains(randomRecommendation)) {
+        return randomRecommendation;
+      } else {
+        invalidRecommendations.add(randomRecommendation);
+      }
+      // If the user has looked at every visualization, return the first in the history.
+      if (invalidRecommendations.size() >= 6885) {
+        isValid = true;
+      }
+    }
+    return userHistory.get(0);
+  }
+
+  // Takes in a map and updates the key/value pairs
+  private void updateMap(HashMap<String, Integer> frequencyMap, String item) {
+    if (frequencyMap.containsKey(item)) {
+      frequencyMap.put(item, frequencyMap.get(item) + 1);
+    } else {
+      frequencyMap.put(item, 1);
+    }
+  }
+
+  private void getFrequencyMaps(ArrayList<VisualizationData> userHistory) {
+    // calculate the frequencies of each type of query that the user has made
+    for (int i = 0; i < userHistory.size(); i++) {
+      VisualizationData currentElement = userHistory.get(i);
+      updateMap(personTypeMap, currentElement.getPersonType());
+      updateMap(actionMap, currentElement.getAction());
+      updateMap(locationMap, currentElement.getLocation());
+      updateMap(yearsMap, currentElement.getYear());
+    }
+  }
+
+  // Takes in a hashmap and returns a linkedHashMap sorted by value in descending order such that 
+  // the most frequented query types are first in the list
+  private LinkedHashMap<String, Integer> sortHashMapDescending(HashMap<String, Integer> frequencyMap) {
+    LinkedHashMap<String, Integer> reverseSortedMap = new LinkedHashMap<>();
+        frequencyMap.entrySet().stream().sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                .forEachOrdered(x -> reverseSortedMap.put(x.getKey(), x.getValue()));
+    return reverseSortedMap;
+  }
+
+  // Return suggested recommendations based on the history. Create hashmaps for each input for all of the queries,
+  // then create four combinations based on the most frequented fields.
+  private ArrayList<VisualizationData> getRecommendations(ArrayList<VisualizationData> userHistory, String userId) {
+    // Populate the frequency maps for each field
+    getFrequencyMaps(userHistory);
```

Comment:
Not sure exactly where this issue comes from or if this is the expected behavior but the recommendations based on history seem to be exclusively all ages population in each U.S. state for various years. I thought it might've been because there were too many queries already in my history, so I cleared my history and made four queries all with "worked in", but the recommendation were the same. 
![image](https://user-images.githubusercontent.com/20865277/91476196-a45da780-e86a-11ea-8a5d-73ebe99ea7fd.png)

Reply:
Hmm that is weird. I didn't run into that issue. Maybe if you try refreshing the page? Or mvn clean?

Reply:
Yeah after clearing everything it's working great! Thanks